pkg/bpf/bpffs.go

```diff
@@ -210,12 +197,22 @@ func mountFS() error {
 		}
 
 	}
-	if !IsBpffs(mapRoot) {
-		// TODO currently on minikube isBpffs check is failing. We need to make the following log
-		// fatal again. This will be tracked in #Issue 1475
-		//log.WithField(logfields.Path, mapRoot).Fatal("BPF: path is not mounted as a BPF filesystem.")
-		log.WithField(logfields.Path, mapRoot).Debug("BPF: path is not mounted as a BPF filesystem.")
+
+	var fsdata unix.Statfs_t
+	if err := unix.Statfs(mapRoot, &fsdata); err != nil {
+		return fmt.Errorf("BPF filesystem path %s is not mounted", mapRoot)
+	}
+
+	// This is the value of the BPF Filesystem defined in
+	// uapi/linux/magic.h The magic value can potentially be misleading if
+	// the BPF filesystem is mounted in the host and then volume mapped
+	// into a container.
+	magic := uint32(0xCAFE4A11)
+	if uint32(fsdata.Type) != magic {
+		log.WithField(logfields.Path, mapRoot).Warning("BPF root is not a BPF filesystem (%d != %d)",
```

Comment:
Should this be a `Log.Fatal` instead, now that we can detect that the mounted BPF file system is not of kind BPF?

Reply:
Actually how is this different from the `IsBpffs` logic we had before? I am a bit confused?

Reply:
It was previously casting to `int32` while `0xCAFE4A11` requires an `uint32`

```
-	magic := uint32(0xCAFE4A11)
-	var fsdata unix.Statfs_t
-	if err := unix.Statfs(path, &fsdata); err != nil {
-		log.WithField(logfields.Path, path).Error("BPF filesystem path is not mounted")
-		return false
-	}
-	return int32(magic) == int32(fsdata.Type)
```

I'll push another commit which changes this to `stat` and we can make it fail hard as it works in container environments.